lib/pdf/elements/list.rb

```diff
@@ -0,0 +1,18 @@
+class Pdf::Elements::List
+  def self.render(pdf, element, indent: 0)
+    ordered = /ol/ =~ element.name
+
+    index = 0
+    pdf.indent indent do
+      element.css("> li").each do |li|
+        inner_list = li.children.css("ul,ol")
```

Comment:
Yes, you are right, I don't think that example you provided will happen on markdown, but we have a similar scenario with nested lists, I will change the css selector to match only the immediate child